src/functions.js

```diff
@@ -1,148 +1,215 @@
+        })
+        return leng;                           another method with loop                           */
+
+    var numbOfLetters = words.map(function(word) {
+        return word.length
+    });
+    return numbOfLetters;
 }
 
-function cities( capitals, formatter ){
-  // capitals is an array of objects that have a city and country property
-  // for example
-  // {
-  //   city: 'Paris',
-  //   country: 'France'
-  // }
-  // formatter is a function that transforms a capital object into a sentence returns it
-  // such as 'Paris is the capital of France'
-
-  // apply formatter to each object in capitals array and return an array of resulting sentences
+function cities(capitals, formatter) {
+    // capitals is an array of objects that have a city and country property
+    // for example
+    // {
+    //   city: 'Paris',
+    //   country: 'France'
+    // }
+    // formatter is a function that transforms a capital object into a sentence returns it
+    // such as 'Paris is the capital of France'
+
+    // apply formatter to each object in capitals array and return an array of resulting sentences
+
+    var transform = capitals.map(function formatter(item) {
+        return (item.city + ' is the capital of ' + item.country);
+    });
+    return transform
 }
 
-function largerThanTen( numbers ){
-  // numbers is an array of numbers
-  // return a new array that contains only numbers from the input array which are greater than 10
-  // hint: filter
+
+function largerThanTen(numbers) {
+    // numbers is an array of numbers
+    // return a new array that contains only numbers from the input array which are greater than 10
+    // hint: filter
+
+    var higherThanTen = numbers.filter(function(number) {
+        return number > 10;
+    });
+    return higherThanTen;
 }
 
-function even( numbers ){
-  // numbers is an array of numbers
-  // return a new array that contains only even numbers from the input array
-  // hint: you may want to use the modulus operator '%'
+function even(numbers) {
+    // numbers is an array of numbers
+    // return a new array that contains only even numbers from the input array
+    // hint: you may want to use the modulus operator '%'
+
+    var onlyEven = numbers.filter(function(item) {
+        return item % 2 === 0;
+    });
+    return onlyEven;
 }
 
-function findTheNeedle( words ){
-  // words is an array of words
-  // return the index of the word 'needle'
-  // hint: indexOf
+function findTheNeedle(words) {
+    // words is an array of words
+    // return the index of the word 'needle'
+    // hint: indexOf
+
+    return words.indexOf('needle');
 }
 
-function findLargest( numbers ){
-  // numbers is an array of numbers
-  // return the largest number from that array
+function findLargest(numbers) {
+    // numbers is an array of numbers
+    // return the largest number from that array
+
+    var sorted = numbers.sort(function(a, b) { return a - b });
+    return sorted.pop();
```

Comment:
This works and it's a good answer. However the sort() method sorts the input which might cause problems in real world programming, so we tend to avoid modifying the input of a function to create what we call a "pure function" instead. There is an interesting article here if you're interested why: https://medium.com/@jamesjefferyuk/javascript-what-are-pure-functions-4d4d5392d49c

You can use the method `Math.max` to avoid mutating the input.

```
return Math.max(numbers);
```